tomox/order_processor.go

```diff
@@ -541,3 +543,67 @@ func DoSettleBalance(coinbase common.Address, takerOrder, makerOrder *tomox_stat
 	tomox_state.SetTokenBalance(makerExOwner, newMakerFee, makerOrder.QuoteToken, statedb)
 	return nil
 }
+
+func (tomox *TomoX) ProcessCancelOrder(tomoXstatedb *tomox_state.TomoXStateDB, statedb *state.StateDB, chain consensus.ChainContext, coinbase common.Address, orderBook common.Hash, order *tomox_state.OrderItem) (error, bool) {
+	exRelayerFee := tomox_state.GetExRelayerFee(order.ExchangeAddress, statedb)
+	if exRelayerFee.Cmp(common.RelayerCancelFee) < 0 {
+		log.Debug("Relayer not enough fee when cancel order", "relayer fee ", exRelayerFee)
+		return nil, true
+	}
+	baseTokenDecimal, err := tomox.GetTokenDecimal(chain, statedb, coinbase, order.BaseToken)
+	if err != nil || baseTokenDecimal.Sign() == 0 {
+		log.Debug("Fail to get tokenDecimal ", "Token", order.BaseToken.String(), "err", err)
+		return err, false
+	}
+	var makerBalance *big.Int
+	switch order.Side {
+	case tomox_state.Ask:
+		makerBalance = tomox_state.GetTokenBalance(order.UserAddress, order.BaseToken, statedb)
```

Comment:
:+1: 
in this case, user has baseToken, it makes sense to charge cancelFee in baseToken